dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/ManageGuestbooksPage.java

```diff
@@ -273,13 +172,12 @@ private void saveDataverse(String successMessage, String failureMessage) {
         if (failureMessage.isEmpty()) {
             failureMessage = "dataset.manageGuestbooks.message.editFailure";
         }
-        try {
-            engineService.submit(new UpdateDataverseCommand(getDataverse(), null, null, dvRequestService.getDataverseRequest(), null));
-            JsfHelper.addFlashSuccessMessage(BundleUtil.getStringFromBundle(successMessage));
-        } catch (CommandException ex) {
-            JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle(failureMessage));
-        }
 
+        String finalSuccessMessage = successMessage;
+        String finalFailureMessage = failureMessage;
+        Try.of(() -> manageGuestbooksService.createOrUpdate(dataverse))
+                .onSuccess(dv -> JsfHelper.addFlashSuccessMessage(BundleUtil.getStringFromBundle(finalSuccessMessage)))
+                .onFailure(throwable -> JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle(finalFailureMessage)));
```

Comment:
I would get rid of method `private void saveDataverse(String successMessage, String failureMessage)` and execute this try separate in `enableGuestbook` and `disableGuestbook` and get rid of `public saveDataverse` it is not used